src/charm.py

```diff
@@ -234,6 +242,20 @@ def _login_ui_layer(self) -> Layer:
         }
         return Layer(pebble_layer)
 
+    def _get_tracing_endpoint_info_http(self) -> str:
```

Comment:
Nitpick:

For both of the `_get_tracing_endpoint_info_http` and `_get_tracing_endpoint_info_grpc`, I'd prefer to return early for error/unhappy conditions:

```python
default_otlp_endpoint = ""
if not self.model.relations[self._tracing_relation_name]:
    return default_otlp_endpoint

return self.tracing.otlp_grpc_endpoint or default_otlp_endpoint
```